pdfminer/pdffont.py

```diff
@@ -706,6 +697,36 @@ def __init__(self, rsrcmgr, spec, strict=settings.STRICT):
         PDFFont.__init__(self, descriptor, widths, default_width=default_width)
         return
 
+    def cmap_setter(self, spec, strict):
+        """
+        For certain PDFs, Encoding Type isn't mentioned as an attribute of
+        Encoding but as an attribute of CMapName, where CMapName is an
+        attribure of spec['Encoding'].
+        The horizaontal/vertical modes are mentioned with diffrent name
+        such as 'DLIdent-H/V','OneByteIdentityH/V','Identity-H/V'
+        """
+        try:
+            spec_encoding = spec['Encoding']
+            if hasattr(spec_encoding, 'name'):
+                cmap_name = literal_name(spec['Encoding'])
+            else:
+                cmap_name = literal_name(spec_encoding['CMapName'])
+        except KeyError:
+            if strict:
+                raise PDFFontError('Encoding is unspecified')
+            cmap_name = 'unknown'
+        if type(cmap_name) is PDFStream:
```

Comment:
This is never true since `literal_name` always returns a `str`. 

Remove this if clause?

Reply:
```
def literal_name(x):
    if not isinstance(x, PSLiteral):
        if settings.STRICT:
            raise PSTypeError('Literal required: %r' % (x,))
        else:
            name=x
    else:
        name=x.name
        if six.PY3:
            try:
                name = str(name,'utf-8')
            except:
                pass
    return name
```

```
    def get_cmap_from_spec(self, spec, strict):
        try:
            spec_encoding = spec['Encoding']
            if hasattr(spec_encoding, 'name'):
                cmap_name = literal_name(spec['Encoding'])
            else:
                cmap_name = literal_name(spec_encoding['CMapName'])
        except KeyError:
            if strict:
                raise PDFFontError('Encoding is unspecified')
            cmap_name = 'unknown'
        if type(cmap_name) is PDFStream:
            if 'CMapName' in cmap_name:
                cmap_name = cmap_name.get('CMapName').name
            else:
                if strict:
                    raise PDFFontError('CMapName unspecified for encoding')
                cmap_name = 'unknown'
        if cmap_name in IDENTITY_ENCODER:
            return CMapDB.get_cmap(cmap_name)
        else:
            return CMap()
```
> This is never true since literal_name always returns a str

If `x` is not an instance of `PSLiteral` it will return the `__repr__`(if __repr__ exist). `__repr__` function actually doesn't return a type str. So the name.replace operation isn't working for such cases.

In case the cmap_name is not an instance of PSLiteral but an instance of PDFStream, we would try to extract `cmap_name`, so `if type(cmap_name) is PDFStream` is helpful for this case. 

With regards to PDFs mentioned in #210. Those PDF are being extracted fine with no errors.

> For example, earlier 'DLIdentit-H' was mapped to 'Identity-H' which returned IdentityCMap(WMode=0) from CMapDB.get_cmap(). With this PR it returns CMap().

I have mapped DLIdentit-H intentionally with CMap() for the following reasons. 
- The PR @jtkese raised solved the issue by using `.get` method to extract cmap_name when the instance type is `PDFStream`.
- I haven't found any conclusive evidence of considering same mapping for DLIdentit-H and Identity-H (documentations available online). Also I haven't seen DLIdentit-H as supported encoding.

Reply:
> If x is not an instance of PSLiteral it will return the __repr__(if repr exist). __repr__ function actually doesn't return a type str. 

That seems strange to me. __repr__ should [always](https://docs.python.org/3/reference/datamodel.html#object.__repr__) return a string. But I guess we can leave it as it is for now. Could you point out where this is happening? I cannot find any related use of repr() or __repr__() in the repo.

> I haven't found any conclusive evidence of considering same mapping for DLIdentit-H and Identity-H 

Good enough. Let's use the default of `CMap()` when we do not know what to do!

Reply:
https://github.com/pdfminer/pdfminer.six/blob/master/pdfminer/pdftypes.py#L201
```
    def __repr__(self):
        if self.data is None:
            assert self.rawdata is not None
            return '<PDFStream(%r): raw=%d, %r>' % (self.objid, len(self.rawdata), self.attrs)
        else:
            assert self.data is not None
            return '<PDFStream(%r): len=%d, %r>' % (self.objid, len(self.data), self.attrs)
```
> That seems strange to me. __repr__ should always return a string.

Yes, I have also assumed that the string operations are valid for `__repr__` before coming through this bug. This being the very reason many developers use `__str__` data model for denoting class object.

Reply:
That sounds strange to me! In which cases does this `__repr__()` returns something other than a string? If you could provide an example than we can open a new super specific issue.

Reply:
> In which cases does this `__repr__()` returns something other than a string? If you could provide an example than we can open a new super specific issue.

I feel it's depending on PDFs. Let me see some PDF that have been reported by users as incompatible.